### PR TITLE
RUM-5770 [Mobile Session Replay] Update documentation with Fine-Grained Masking

### PR DESCRIPTION
**Pull request by jonathanmos**
### What does this PR do? What is the motivation?
Adds documentation to support the upcoming release of Fine-Grained privacy levels on Mobile RUM Session Replay. Also adds documentation regarding the new `startRecording`/`stopRecording`/`startRecordingImmediately` APIs.

### Merge instructions
On Android this was released in version 2.14.0 on the 25/9. 
On ios this was released in version 2.18.0 on 26/9.

- [X] Please merge after reviewing

### CODE REVIEW

**Comment by buraizu** [comment]
Thanks, created DOCS-9026 to review

File: content/en/real_user_monitoring/session_replay/mobile/privacy_options.md [hunk 1]
```diff
@@ -30,18 +30,283 @@ By enabling Mobile Session Replay, you can automatically mask sensitive elements
 
 ## Configuring masking modes
 
-Using the masking modes below, you can override the default setup on a per-application basis.
+## Fine-Grained Masking
+Using the masking modes below, you can override the default setup on a per-application basis. Masking is fine-grained — you can override masking for text and inputs, images, and touches individually to create a custom configuration that suits your needs.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/session_replay/mobile/privacy_options.md:34]:
```suggestion
Using the masking modes below, you can override the default setup on a per-application basis. Masking is fine-grained, which means you can override masking for text, text inputs, images, and touches individually to create a custom configuration that suits your needs. 
```

File: content/en/real_user_monitoring/session_replay/mobile/privacy_options.md [hunk 2]
```diff
@@ -30,18 +30,283 @@ By enabling Mobile Session Replay, you can automatically mask sensitive elements
...
+{{< /code-block >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+#### Mask content images
+To manage content masking while still showing system images, users can choose the following options:
+
+On iOS, users can select the `mask_non_bundled_only` setting, which replaces any image that is not part of the system with a "Content Image" placeholder.
+
+On Android, users can select the `mask_large_only`  setting, which replaces images whose dimensions exceed 100x100dp with a "Content Image" placeholder. Note that these dimensions refer to the drawable resource, not the view's size.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/session_replay/mobile/privacy_options.md:169]:
```suggestion
On Android, users can select the `mask_large_only` setting, which replaces images with dimensions that exceed 100x100dp with a "Content Image" placeholder. 

**Note**:  that these dimensions refer to the drawable resource, not the view's size.
```

File: content/en/real_user_monitoring/session_replay/mobile/privacy_options.md [hunk 3]
```diff
@@ -30,18 +30,283 @@ By enabling Mobile Session Replay, you can automatically mask sensitive elements
...
+
+{{< /code-block >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+### Touch masking
+By default, the `hide` setting is enabled for all touches. With this setting enabled, all touches on screen are hidden.
+
+#### Hide all touches
+With the `hide` setting enabled, all touches that occur during the replay will be hidden. This is the default setting.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/session_replay/mobile/privacy_options.md:240]:
```suggestion
With the `hide` setting enabled, all touches that occur during the replay are hidden. This is the default setting. 
```

File: content/en/real_user_monitoring/session_replay/mobile/privacy_options.md [hunk 4]
```diff
@@ -30,18 +30,283 @@ By enabling Mobile Session Replay, you can automatically mask sensitive elements
...
+        touchPrivacyLevel: .hide
+    )
+    SessionReplay.enable(with: sessionReplayConfig)
+
+{{< /code-block >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+#### Show all touches
+With the `show` setting enabled, all touches that occur during the replay will be shown.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/session_replay/mobile/privacy_options.md:269]:
```suggestion
With the `show` setting enabled, all touches that occur during the replay are shown. 
```

File: content/en/real_user_monitoring/session_replay/mobile/privacy_options.md [hunk 5]
```diff
@@ -30,18 +30,283 @@ By enabling Mobile Session Replay, you can automatically mask sensitive elements
...
+        touchPrivacyLevel: .show
+    )
+    SessionReplay.enable(with: sessionReplayConfig)
+
+{{< /code-block >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+## Legacy masking - Deprecated
+Please note that this masking API is deprecated and will be removed in a future version. Users are encouraged to migrate to the fine-grained masking options described above.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/session_replay/mobile/privacy_options.md:298]:
```suggestion
This masking API is deprecated. Users are encouraged to migrate to the fine-grained masking options described above.
```

File: content/en/real_user_monitoring/session_replay/mobile/privacy_options.md [hunk 6]
```diff
@@ -70,7 +335,7 @@ By default, the `mask` setting is enabled for all data. With this setting enable
 
 ### Mask only input elements
 
-With the `mask user input` setting enabled, any input field is replaced with anonymized text. 
+With the `mask user input` setting enabled, any input field is replaced with anonymized text. Note that in addition to this, touches are hidden, and some images (>100x100dp images on android/non-system images on ios) are replaced with placeholders.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/session_replay/mobile/privacy_options.md:338]:
```suggestion
With the `mask user input` setting enabled, any input field is replaced with anonymized text.

**Note**: In addition to this behavior, touches are hidden, and some images (>100x100dp images on Android/non-system images on iOS) are replaced with placeholders.
```

File: content/en/real_user_monitoring/session_replay/mobile/setup_and_configuration.md [hunk 7]
```diff
@@ -157,6 +157,37 @@ var sessionReplayConfig = SessionReplay.Configuration(
 {{% /tab %}}
 {{< /tabs >}}
 
+### Start the recording manually
+
+By default, Session Replay will start recording automatically. However, if you prefer to manually start recording at a specific point in your application, you can use the optional `startRecordingImmediately` parameter as shown below, and later call `SessionReplay.startRecording()`.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/session_replay/mobile/setup_and_configuration.md:162]:
```suggestion
By default, Session Replay starts recording automatically. However, if you prefer to manually start recording at a specific point in your application, you can use the optional `startRecordingImmediately` parameter as shown below, and later call `SessionReplay.startRecording()`.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Some minor feedback to adhere to doc style guidelines, but otherwise looks great. Thanks for this PR!

**Comment by mariedm** [comment]
Thanks for the review @rtrieu!

File: content/en/real_user_monitoring/session_replay/mobile/privacy_options.md (hunk 1 above)
Comment (jonathanmos) [reply content/en/real_user_monitoring/session_replay/mobile/privacy_options.md:34]: additionally, the privacy setting for text and inputs is a single setting so dividing them here may be confusing to customers who would expect separate apis for each one.

Comment (rtrieu) [reply content/en/real_user_monitoring/session_replay/mobile/privacy_options.md]: @mariedm and @jonathanmos thank you both for the clarification. I've changed my suggestion back to the original text but kept "which means"

**Review by rtrieu — CHANGES_REQUESTED** [review]
LGTM

**Review by rtrieu — APPROVED** [review]
approved

**Pull request merged**